File: Data Structures & Algorithms/contains-duplicate-ii/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {boolean}
     */
    containsNearbyDuplicate(nums, k) {
        let set = new Set()

        let left = 0;
        let right = left;

        while (right < nums.length) {
            if (right - left <= k) {
                if (set.has(nums[right])) {
                    return true;
                } else {
                    set.add(nums[right])
                    right++
                }
            } else {
                set.delete(nums[left])
                left++
            }
        }

        return false
    }
}

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-1.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int min = Integer.MAX_VALUE;

        int left = 0, right = 0;

        int sum = 0;
        while(right < nums.length) {
            
            sum += nums[right++];

            while(sum >= target) {
                min = Math.min(min, right - left);
                sum -= nums[left++];
            }
        }


        return min == Integer.MAX_VALUE ? 0 : min; 
    }
}